### PR TITLE
Simplified TestHelper

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds a simplified version of TestHelper in namespace TestNew (a temporary name), and updates the Ice/acm test to use this new TestHelper.

### CODE REVIEW

File: csharp/test/TestCommon/TestHelper.cs [hunk 1]
```diff
@@ -284,3 +284,260 @@ public static async Task<int> RunTestAsync<T>(string[] args) where T : TestHelpe
...
+            }
+
+            return new Communicator(properties, tlsServerOptions: tlsServerOptions, observer: observer);
+        }
+
+        public string GetTestEndpoint(int num = 0, string transport = "", bool ephemeral = false) =>
+            GetTestEndpoint(Communicator.GetProperties(), num, transport, ephemeral);
+
+        public static string GetTestEndpoint(
+            Dictionary<string, string> properties, int num = 0, string transport = "", bool ephemeral = false)
```
Comment (pepone) [inline csharp/test/TestCommon/TestHelper.cs:346]: This uses the old style, should be one param per line

File: csharp/test/TestCommon/TestHelper.cs [hunk 2]
```diff
@@ -284,3 +284,260 @@ public static async Task<int> RunTestAsync<T>(string[] args) where T : TestHelpe
...
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidConfigurationException(
+                    $"invalid value for for Test.Protocol: `{value}'", ex);
+            }
+        }
+
+        public static ZeroC.Ice.Encoding GetTestEncoding(Dictionary<string, string> properties)
+            => ProtocolExtensions.GetEncoding(GetTestProtocol(properties));
```
Comment (pepone) [inline csharp/test/TestCommon/TestHelper.cs:474]:
```suggestion
        public static ZeroC.Ice.Encoding GetTestEncoding(Dictionary<string, string> properties) =>
            ProtocolExtensions.GetEncoding(GetTestProtocol(properties));
```

File: csharp/test/TestCommon/TestHelper.cs [hunk 3]
```diff
@@ -284,3 +284,260 @@ public static async Task<int> RunTestAsync<T>(string[] args) where T : TestHelpe
...
+            GetTestProxy(identity, Communicator.GetProperties(), num, transport);
+
+        public abstract Task RunAsync(string[] args);
+
+        public static async Task<int> RunTestAsync<T>(Communicator communicator, string[] args)
+            where T : TestHelper, new()
+        {
+            try
+            {
+                var helper = new T() { Communicator = communicator };
```
Comment (pepone) [inline csharp/test/TestCommon/TestHelper.cs:528]: That's nice, I was surprised when I see the init method was private, and the abstract class has no constructor.

**Review by pepone — APPROVED** [review]

File: csharp/test/TestCommon/TestHelper.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/test/TestCommon/TestHelper.cs:474]: I think we should just add a `using ZeroC.Ice;`

Comment (pepone) [reply csharp/test/TestCommon/TestHelper.cs:474]: Fine, I was just pointing that the `=>` is out of place

**Review by externl — APPROVED** [review]

**Comment by externl** [comment]
Should TestHelper constructer just always call `Communicator.ActivateAsync()` instead of it being the first thing you need to do in the test?

**Comment by bernardnormier** [comment]
> Should the `TestHelper` constructor just always call `Communicator.ActivateAsync()` instead of it being the first thing you need to do in the test?

No, because we can't (shouldn't) call Async methods from a constructor. And even if we could, some tests may need to perform setup between the communicator's construction and its activation.

**Comment by externl** [comment]
> > Should the `TestHelper` constructor just always call `Communicator.ActivateAsync()` instead of it being the first thing you need to do in the test?
> 
> No, because we can't (shouldn't) call Async methods from a constructor. And even if we could, some tests may need to perform setup between the communicator's construction and its activation.

Good point

File: csharp/test/TestCommon/TestHelper.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/test/TestCommon/TestHelper.cs:346]: Fixed.

File: csharp/test/TestCommon/TestHelper.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/test/TestCommon/TestHelper.cs:474]: Fixed.

File: csharp/test/TestCommon/TestHelper.cs (hunk 3 above)
Comment (bernardnormier) [reply csharp/test/TestCommon/TestHelper.cs:528]: What is less nice is:
```
public Communicator Communicator { get; private init; } = null!;
```
without this ` = null!` we get a warning.

Comment (pepone) [reply csharp/test/TestCommon/TestHelper.cs:528]: is it the same if you make it `protected init`?

Comment (bernardnormier) [reply csharp/test/TestCommon/TestHelper.cs:528]: Yes.

**Pull request merged**